### PR TITLE
Fix warning, update to newer OCaml/dune, remove Rresult dependency

### PR DESCRIPTION
**Pull request by vitalydolgov**
Hi! I propose some changes to the library (v. 0.1.5), namely 

 * fixed warning 31 (from 4.03 it's handled as error)
 * fixed deprecation warning from ocplib-endian
 * updated to OCaml 4.08 and dune 2.9
 * removed Rresult dependency

The most important change is the first one, as on the modern compiler the library links with warning due to conflict with Types: 
```
Warning 31: files src/core/osc.cma(Types) and /home/vitaly/.opam/4.11.1/lib/ocaml/compiler-libs/ocamlcommon.cma(Types) both define a module named Types
```

### CODE REVIEW

File: test/unix/dune [hunk 1]
```diff
@@ -8,8 +8,8 @@
   (modules test_interop_sclang)
   (libraries osc-unix test_common))
 
-(alias
-  (name   runtest)
+(rule
```
Comment (zapashcanon) [inline test/unix/dune:11]: it should be possible to convert this to a proper `test` stanza ?

Comment (vitalydolgov) [reply test/unix/dune]: Yeah, good idea :+1:

**Comment by johnelse** [comment]
Thanks, this is really useful! I'm not sure about the last commit though - `test_unix` and `test_interop_sclang` are independent tests, I don't think there should be dependencies between them.

**Comment by vitalydolgov** [comment]
We can split them, i.e. make directory `interop_supercollider` and move the test in question there. What do you think?

**Comment by johnelse** [comment]
Sure, moving `test_interop_sclang` to a new directory would be OK. I think the dune syntax you had originally, i.e.

```
(rule
  (alias runtest)
  (deps
    (:x test_interop_sclang.exe)
    (source_tree helpers))
  (action (run ./helpers/test-interop-sclang.sh)))
```

was fine though, that's pretty close to what the dune docs suggest: https://dune.readthedocs.io/en/stable/dune-files.html#alias

**Comment by vitalydolgov** [comment]
I've separated tests. To my preference though, to have all tests in one dune is more convenient than to separate them while there only one module per test, maybe you consider reorganization in future.

**Comment by johnelse** [comment]
I'm not against all the tests being in one directory, IIRC I tried that first but it was easier to get everything to build if I split them up. I'll merge this now though as this is a definite improvement, thanks for the PR!

**Pull request merged**